### DISCUSSION TITLE
Derived atom with unresolved promise is not re-calculated

### DISCUSSION THREAD

**Discussion by hyp3rflow** [pmndrs/jotai#1465] [Q&A] [answered]
Hi. I want to trigger suspense when the atom is initialized.
I found that I can create derived default atom using 3 atoms. https://github.com/pmndrs/jotai/issues/352#issuecomment-797829152, so I thought if I use unresolved promise for defaultValueAtom's value (like `defaultValueAtom = atom(new Promise(() => {}))`)

So I wrote some code like below:
```javascript
const defaultValueAtom = atom(() => new Promise(() => { }));
const overwrittenValueAtom = atom((get) => {
  return get(flagAtom) ? 'done' : undefined;
});
const valueAtom = atom((get) => {
  return get(overwrittenValueAtom) || get(defaultValueAtom);
});

const flagAtom = atom(false);

function Component() {
  useAtom(valueAtom);
}
function Page() {
  const setFlag = useSetAtom(flagAtom);
  return <>
    <button onClick={() => setFlag(true)} />
    <Suspense fallback={<div>loading</div>}>
      <Component />
    </Suspense>
  </>
}
```

But it seems that although I clicked the button to set `flagAtom` true, `valueAtom` wasn't recalculated and the `Suspense` never resolved.
I don't know how exactly the `Suspense` works and how `Suspense` is triggered in jotai internally, but is this intended behavior?
If so, how can I trigger the `Suspense` for loading intentionally using derived atom?

Thanks :)

**Comment by dai-shi** [comment, marked as answer]
We've discussed about this some time ago, and while this seems like a desirable behavior, it's not currently possible. It's probably because how Suspense works.
So, it's an intended behavior at least for now.
What you could do is to write directly to the `defaultValueAtom`.

**Reply by hyp3rflow** [reply]
Thanks for clarifying! 😊
